l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -1163,10 +1163,68 @@ def set_welfares_fund(
             return
 
         WelfareFundLineModel = self.env['welfare.fund.data.line']
+
+        wts = False
+        if any([welfareLine for welfareLine in Welfares
+                if welfareLine.TipoCassa == 'TC07']):
+            # Search the matching withholding tax
+            wts = self.env['withholding.tax'].search(
+                [('wt_types', '=', 'enasarco')])
+
+        processedLines = []
         for welfareLine in Welfares:
-            WalferLineVals = self._prepareWelfareLine(
-                invoice.id, welfareLine)
+            # Handle only TC07
+            if welfareLine.TipoCassa == 'TC07':
+                if not wts:
+                    raise UserError(_(
+                        "The bill contains Welfare Fund tax with "
+                        "Type %s, "
+                        "but such a tax is not found in your system. Please "
+                        "set it."
+                    ) % welfareLine.TipoCassa)
+
+                wt_found = False
+                for wt in wts:
+                    if wt.tax == float(welfareLine.AlCassa):
+                        wt_found = wt
+                        break
+
+                if not wt_found:
+                    raise UserError(_(
+                        "The bill contains Welfare Fund tax with "
+                        "Type %s and Tax %s "
+                        "but such a tax is not found in your system. Please "
+                        "set it."
+                    ) % (welfareLine.TipoCassa, float(welfareLine.AlCassa)))
+
+            WalferLineVals = self._prepareWelfareLine(invoice.id, welfareLine)
             WelfareFundLineModel.create(WalferLineVals)
+
+            # We are not interested to continue if Enasarco is used
+            if welfareLine.TipoCassa == 'TC07':
+                found = False
+                for line in invoice.invoice_line_ids:
+                    # search an invoice line having the same amount of the
+                    # ImportoContributoCassa
+                    if (line.id not in processedLines and
+                            line.invoice_line_tax_wt_ids):
+                        processedLines.append(line.id)
+                        ids = line.invoice_line_tax_wt_ids.ids
+                        ids.append(wt_found.id)
+                        line.update({
+                            'invoice_line_tax_wt_ids': [(6, 0, ids)],
```

Comment:
Qui sovrascrivi `invoice_line_tax_wt_ids` aggiungendo `wt_found`? Cosa potrebbe esserci già in `invoice_line_tax_wt_ids`? Perchè è necessario aggiungerlo se già impostato da `set_withholding_tax`?

Reply:
@eLBati Serve solo a garantire che nel caso sia già calcolata una ritenuta, questa venga preservata

Reply:
Non capisco: scrivi `wt_found` solo se `invoice_line_tax_wt_ids` è già valorizzato ?

Più in generale, se l'obiettivo di questo override è aggiungere la ritenuta di tipo enasarco, perchè non può limitarsi a fare solo quello? Se nell'XML c'è, viene aggiunta la ritenuta